src/gui/PageView.cpp

```diff
@@ -498,8 +515,16 @@ auto XojPageView::onButtonReleaseEvent(const PositionInputData& pos) -> bool {
             }
         }
 
-        delete this->inputHandler;
-        this->inputHandler = nullptr;
+		ToolHandler* h = control->getToolHandler();
+		if (h->getDrawingType()==DRAWING_TYPE_SPLINE)
+		{
+			// Do NOT delete inputHandler yet, since drawing is not finished
+		} 
+		else 
+		{ 
```

Comment:
Nitpick: use `if (h->getDrawingType() != DRAWING_TYPE_SPLINE)` and change the comment to indicate that the spline tool manually finalizes drawing.